core-java-modules/core-java-lang-5/src/main/java/com/baeldung/generics/Cat.java

```diff
@@ -0,0 +1,30 @@
+package com.baeldung.generics;
+
+class Cat extends Animal implements Comparable<Cat> {
+    public Cat(String type, String name) {
+        super(type, name);
+    }
+
+    @Override
+    public String makeSound() {
+        return "Meow";
+    }
+
+    @Override
+    public int compareTo(Cat cat) {
+        return this.getName().length() - cat.getName().length();
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("Cat{");
+        sb.append("type='")
+            .append(type)
+            .append('\'');
+        sb.append(", name='")
+            .append(name)
+            .append('\'');
+        sb.append('}');
+        return sb.toString();
```

Comment:
Since the string can be done in on line we are not gaining performance for using StringBuilder so I think just appending the literals and variables would make the code easier to read.

`return "Cat{" + "type='" + type + '\'' + ", name='" + name + '\'' + '}';`